packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -110,10 +166,50 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
   }
 
   public async apply(sdk: ISDK): Promise<any> {
-    return sdk.lambda().updateFunctionCode({
-      FunctionName: this.lambdaFunctionResource.physicalName,
-      S3Bucket: this.lambdaFunctionResource.code.s3Bucket,
-      S3Key: this.lambdaFunctionResource.code.s3Key,
-    }).promise();
+    const lambda = sdk.lambda();
+    const resource = this.lambdaFunctionResource.resource;
+
+    if (resource.code !== undefined) {
+      await lambda.updateFunctionCode({
+        FunctionName: this.lambdaFunctionResource.physicalName,
+        S3Bucket: resource.code.s3Bucket,
+        S3Key: resource.code.s3Key,
+      }).promise();
```

Comment:
Any reason we're waiting here for the Lambda code to update first before handling the tags? Perhaps we can do these in parallel?

Reply:
I'm happy to, if you insist. I don't think it's a big deal -- tag updates are _fast_ -- but if you can walk me through what that code should look like I will (I don't fully grok how parallel code in JS works, sorry to say).

Reply:
Sure. Basically, use `Promise.all()` instead of waiting for each promise before creating the new one.